webhook_server/tests/test_ai_review.py

```
@@ -0,0 +1,219 @@
+            mock_github_webhook.logger.error.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_unexpected_exception(self, mock_github_webhook: Mock, mock_pull_request: Mock) -> None:
+        with patch("webhook_server.libs.ai_review.httpx.AsyncClient", side_effect=RuntimeError("boom")):
+            await call_ai_reviewer(mock_github_webhook, mock_pull_request, trigger="pr-opened")
+        mock_github_webhook.logger.exception.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_cancelled_error_reraised(self, mock_github_webhook: Mock, mock_pull_request: Mock) -> None:
+        with patch("webhook_server.libs.ai_review.httpx.AsyncClient", side_effect=asyncio.CancelledError):
+            with pytest.raises(asyncio.CancelledError):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)